src/main/java/dev/suse/explore/cap/CloudFoundryAPI.java

```diff
@@ -37,6 +44,35 @@ public boolean userAlreadyExists(String email) {
 	public String buildEnvironmentForUser(String email) throws Exception{
 		CloudFoundryOperations ops = context.getBean(CloudFoundryOperations.class);
 		
+		//**********create the org***************
+		//create org name from email: foo.bar@bar.com -> foor_bar_bar_com
+		String orgname = email.replace("\w", "_");
+
+		//How do I tell which parameters the request needs?
+		//TODO: replace quota name string with an env variable or something
+		CreateOrganizationRequest req = CreateOrganizationRequest.builder().organizationName(orgname).quotaDefinitionName("sandbox").build();
+		//Do I need to call .block() in the end? Why if so?
+		ops.organizations().create(req).block();
+
+		//**********make user org manager and developer*************
+		SetOrganizationRoleRequest req = SetOrganizationRoleRequest.builder().organizationName(orgname).organizationRole("Manager").build();
+		ops.useradmin().setOrganizationRole(req).block();
+		SetOrganizationRoleRequest req = SetOrganizationRoleRequest.builder().organizationName(orgname).organizationRole("Developer").build();
+		ops.useradmin().setOrganizationRole(req).block();
+
+		//***********create the default spaces************
+		//TODO: there is probably an elegant way to fold these into one call somehow
+		//TODO: Does the user get roles assigned for these spaces automatically?
```

Comment:
In the userAdmin() set of functions I think I saw role management?

Reply:
Yeah I saw that. I was just wondering if we need to set each and every role manually or if some of then are inferred by CF. If I'm an OrgAdmin and I create a new space, do I automatically become admin for that space?

Reply:
From my very limited experience on the command line, I think I had to create them afterwards.